chapter_preliminaries/linear-algebra.ipynb

```diff
@@ -0,0 +1,1881 @@
+    "and the corresponding implementation in code.\n",
+    "\n",
+    "## Scalars\n",
+    "\n",
+    "If you never studied linear algebra or machine learning,\n",
+    "then your past experience with math probably consisted\n",
+    "of thinking about one number at a time.\n",
+    "And, if you ever balanced a checkbook\n",
+    "or even paid for dinner at a restaurant\n",
+    "then you already know how to do basic things\n",
+    "like adding and multiplying pairs of numbers.\n",
+    "For example, the temperature in Palo Alto is $52$ degrees Fahrenheit.\n",
+    "Formally, we call values consisting\n",
+    "of just one numerical quantity *scalars*.\n",
+    "If you wanted to convert this value to Celsius\n",
+    "(the metric system's more sensible temperature scale),\n",
+    "you would evaluate the expression $c = \\frac{5}{9}(f - 32)$, setting $f$ to $52$.\n",
+    "In this equation, each of the terms---$5$, $9$, and $32$---are scalar values.\n",
+    "The placeholders $c$ and $f$ are called *variables*\n",
+    "and they represent unknown scalar values.\n",
+    "\n",
+    "In this book, we adopt the mathematical notation\n",
+    "where scalar variables are denoted\n",
+    "by ordinary lower-cased letters (e.g., $x$, $y$, and $z$).\n",
+    "We denote the space of all (continuous) *real-valued* scalars by $\\mathbb{R}$.\n",
+    "For expedience, we will punt on rigorous definitions\n",
+    "of what precisely *space* is,\n",
+    "but just remember for now that the expression $x \\in \\mathbb{R}$\n",
+    "is a formal way to say that $x$ is a real-valued scalar.\n",
+    "The symbol $\\in$ can be pronounced \"in\"\n",
+    "and simply denotes membership in a set.\n",
+    "Analogously, we could write $x, y \\in \\{0, 1\\}$\n",
+    "to state that $x$ and $y$ are numbers\n",
+    "whose value can only be $0$ or $1$.\n",
+    "\n",
+    "A scalar is represented by a NDArray with just one element.\n",
+    "In the next snippet, we instantiate two scalars\n",
+    "and perform some familiar arithmetic operations with them,\n",
+    "namely addition, multiplication, division, and exponentiation.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%mavenRepo snapshots https://oss.sonatype.org/content/repositories/snapshots/\n",
+    "%maven org.slf4j:slf4j-api:1.7.26\n",
+    "%maven org.slf4j:slf4j-simple:1.7.26\n",
+    "        \n",
+    "%maven ai.djl:api:0.9.0\n",
+    "%maven ai.djl:basicdataset:0.9.0\n",
+    "\n",
+    "// See https://github.com/awslabs/djl/blob/master/mxnet/mxnet-engine/README.md\n",
+    "// MXNet \n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.9.0\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-backport"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ai.djl.ndarray.*;\n",
+    "NDManager nd = NDManager.newBaseManager();\n",
```

Comment:
```suggestion
    "NDManager manager = NDManager.newBaseManager();\n",
```
nd is the name of python package while NDManager here means totally different thing. It is object that helps us manage the native resource. Let's use manager here.